tasks/generate-history/registerArchaeologists.ts

```diff
@@ -0,0 +1,65 @@
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { BigNumber } from "ethers";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import { generateHistoryConfig } from "./config";
+import { range } from "./helpers";
+
+/**
+ * Gives an archaeologist sarco token, approves the sarco token, and registers the archaeologist.
+ * Error stack traces are left out to keep the logs clean.
+ * @param hre The Hardhat Runtime Environment
+ */
+export async function registerArchaeologists(
+  hre: HardhatRuntimeEnvironment
+): Promise<SignerWithAddress[]> {
+  const accounts = await hre.ethers.getSigners();
+  const count = generateHistoryConfig.archaeologistCount;
+
+  if (count > accounts.length) {
+    throw new Error(`Not enough accounts. Count must be less than or equal to ${accounts.length}`);
+  }
+
+  // Get the contract
+  const diamond = await hre.ethers.getContract("Diamond_DiamondProxy");
+  const archaeologistFacet = await hre.ethers.getContractAt("ArchaeologistFacet", diamond.address);
+  const sarcoToken = await hre.ethers.getContract("SarcoTokenMock");
+
+  console.log("Registering archaeologists...");
+  const archaeologistSigners: SignerWithAddress[] = [];
+  for (let i = 0; i < count; i++) {
+    const account = accounts[i];
+    const minimumDiggingFee = 10;
```

Comment:
Could this be a randomized big number? That might help us test the digging fee filters on the front-end

https://ethereum.stackexchange.com/questions/97067/random-bignumber-in-ethers-js

example in that article looks like it lets you set a max value

Reply:
Same answer from the rewrap interval. I started out making it random then made it a constant to simplify the code.